### PR TITLE
implement ingester state reporting message

### PR DESCRIPTION
**Pull request by floren**

### CODE REVIEW

File: ingest/api.go [hunk 1]
```diff
@@ -10,28 +10,32 @@ package ingest
...
 	//MAJOR API VERSIONS should always be compatible, there just may be additional features
 	API_VERSION_MAJOR uint32 = 0
-	API_VERSION_MINOR uint32 = 5
+	API_VERSION_MINOR uint32 = 6
 )
 
 const (
 	configurationBlockSize          uint32          = 1
 	maxStreamConfigurationBlockSize uint32          = 1024 * 1024 //just a sanity check
+	maxIngestStateSize              uint32          = 1024 * 1024 * 1024
```
Comment (traetox) [inline ingest/api.go:30]: a GIGABYTE?!?!?!

File: ingest/api.go [hunk 2]
```diff
@@ -142,3 +146,70 @@ func ParseCompression(v string) (ct CompressionType, err error) {
...
+	UUID          string
+	Name          string
+	Version       string
+	IP            net.IP //child IP, won't be populated unless in child
+	Entries       uint64
+	CacheState    string
+	CacheSize     uint64
+	Children      map[string]IngesterState
+	Configuration json.RawMessage `json:",omitempty"`
+	Metadata      string
```
Comment (traetox) [inline ingest/api.go:160]: should metadata be a json.RawObject so we can stuff "things" in it?

**Review by traetox — APPROVED** [review]
maaaaybe turn down that max state size...

File: ingest/entryReader.go [hunk 3]
```diff
@@ -337,6 +344,39 @@ headerLoop:
...
+		case INGESTER_STATE_MAGIC:
+			// read length of buffer
+			n, err = io.ReadFull(er.bIO, er.buff[0:4])
+			if err != nil {
+				return err
+			}
+			if n < 4 {
+				return errFailedFullRead
+			}
+			length := binary.LittleEndian.Uint32(er.buff[0:4])
```
Comment (djfritz) [inline ingest/entryReader.go:357]: https://tools.ietf.org/html/draft-newman-network-byte-order-01

network byte order is big endian for what it's worth.

File: ingest/entryReader.go [hunk 4]
```diff
@@ -767,6 +809,10 @@ func (ac ackCommand) encode(b []byte) (n int, flush bool, err error) {
 		binary.LittleEndian.PutUint64(b[4:], ac.val)
 		n += 12
 		flush = true
+	case CONFIRM_INGESTER_STATE_MAGIC:
+		binary.LittleEndian.PutUint32(b, uint32(ac.cmd))
```
Comment (djfritz) [inline ingest/entryReader.go:813]: byte order as above

**Review by djfritz — APPROVED** [review]
LGTM

**Comment by floren** [comment]
PTAL

File: ingest/muxer.go [hunk 5]
```diff
@@ -511,7 +511,7 @@ func (im *IngestMuxer) SetRawConfiguration(c json.RawMessage) {
 	im.ingesterState.Configuration = c
 }
 
-func (im *IngestMuxer) SetMetadata(s string) {
+func (im *IngestMuxer) SetMetadata(s json.RawMessage) {
```
Comment (traetox) [inline ingest/muxer.go:514]: can this function be of the form:
```func (im *IngestMuxer) SetMetadata(obj interface{}) (err error) {
        if obj == nil {
             return
        }
        var msg []byte
        if msg, err = json.Marshal(obj): err != nil {
               return
        }
        m.ingesterState.Metadata = json.RawMessage(msg)
        return
}
```

**Review by traetox — CHANGES_REQUESTED** [review]
don't hate me

**Comment by floren** [comment]
PTYAL

**Review by traetox — APPROVED** [review]
LGTM

**Pull request merged**